### PR TITLE
Always use `ctx.Locale.Tr` inside templates

### PR DESCRIPTION
**Pull request by delvh**
Refactoring performed using
```bash
cd templates/
ag --only-matching --nofilename '[^{()\s]+\.Tr\b' | sed 's/ //g' | uniq --unique | sort -h | uniq --unique

 # Then pipe into vim, execute ':sort u', throw out `ctx.Locale.Tr`, and save as '../bad-instances'

for bad_instance in $(cat ../bad-instances); do
  echo $bad_instance;
  for file in $(ag --literal --files-with-matches "$bad_instance") ; do
    vim "$file" -c ":%s/\(\<\)\?$bad_instance\(\>\)\?/\1ctx.Locale.Tr\2" -c ':wq' ;
  done;
done

# Edge cases that are not covered by above
for file in $(ag --files-with-matches -Q '$ctx.Locale.Tr') ; do vim $file -c ':%s/\$ctx\.Locale\.Tr/ctx\.Locale\.Tr/g' -c ':wq'; done
git checkout da0467e3b -- ./status/500.tmpl
for file in $(ag --files-with-matches -Q '$.rootctx.Locale.Tr') ; do vim $file -c ':%sno/$.rootctx.Locale.Tr/ctx.Locale.Tr/g' -c ':wq'; done
for file in $(ag --files-with-matches -Q  '.rootctx.Locale.Tr') ; do vim $file -c ':%sno/.rootctx.Locale.Tr/ctx.Locale.Tr/g' -c ':wq'; done
for file in $(ag --files-with-matches -Q  '.ctxDatactx.Locale.Tr') ; do vim $file -c ':%sno/.ctxDatactx.Locale.Tr/ctx.Locale.Tr/g' -c ':wq'; done
```
This PR is a corollary to #27227 as that PR removes the unneeded `locale` parameters from subtemplates.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
`500.tmpl` is special, it can't use `ctx` at the moment.

**Comment by delvh** [comment]
Done

File: templates/repo/branch_dropdown.tmpl [hunk 1]
```diff
@@ -50,13 +50,13 @@
 		'enableFeed': {{.root.EnableFeed}},
 		'rssURLPrefix': '{{$.root.RepoLink}}/rss/branch/',
 		'branchURLPrefix': '{{if .branchURLPrefix}}{{.branchURLPrefix}}{{else}}{{$.root.RepoLink}}/{{if $.root.PageIsCommits}}commits{{else}}src{{end}}/branch/{{end}}',
-		'branchURLSuffix': '{{if .branchURLSuffix}}{{.branchURLSuffix}}{{else}}{{if $.root.TreePath}}/{{PathEscapeSegments $.root.TreePath}}{{end}}{{end}}',
+		'branchURLSuffix': '{{if .branchURLSuffix}}{{.branchURLSuffix}}{{else}}{{if ctx.Locale.TreePath}}/{{PathEscapeSegments $.root.TreePath}}{{end}}{{end}}',
```
Comment (wxiaoguang) [inline templates/repo/branch_dropdown.tmpl]: I do not feel safe to use your regexp. What's this change?

And I guess there are more problems.

Comment (delvh) [reply templates/repo/branch_dropdown.tmpl]: Okay, yeah, I can guess what happened here.
Will update my regexp.

Comment (delvh) [reply templates/repo/branch_dropdown.tmpl]: I think I've fixed all edge cases now.

**Review by denyskon — APPROVED** [review]
I didn't go through all files, but it looks good to me in general. I think we should merge, and fix possible issues when they appear. Thanks @delvh

**Comment by delvh** [comment]
Hmm… Is the current CI failure related to my changes or not?
I'm rather leaning towards unrelated.

**Comment by denyskon** [comment]
Looks related to me.... the few tests I looked at refer to some text inside pages, or http errors... looks like broken templates

**Comment by delvh** [comment]
Found the culprit.
Am currently fixing it.

**Comment by delvh** [comment]
Okay, now I am confused:
According to the CI, my changes appear to be responsible for the failure.
However, I cannot reproduce the `500` locally.
I cannot even find a template that is called by the respective method.
So, I have no idea what is going wrong here.
Does anyone else have an idea?

**Comment by denyskon** [comment]
@delvh It's the mail template under `templates/mail/team_invite.tmpl`.
It doesn't seem to work with `ctx.Locale.Tr`. When changing back to `.locale.Tr`, the tests pass

**Review by denyskon — APPROVED** [review]

**Review by silverwind — APPROVED** [review]
lgtm after ci passes

**Comment by delvh** [comment]
CI finally passes.

File: templates/mail/issue/default.tmpl [hunk 2]
```diff
@@ -16,7 +16,7 @@
 </head>
 
 <body>
-	{{if .IsMention}}<p>{{.locale.Tr "mail.issue.x_mentioned_you" .Doer.Name | Str2html}}</p>{{end}}
+	{{if .IsMention}}<p>{{ctx.Locale.Tr "mail.issue.x_mentioned_you" .Doer.Name | Str2html}}</p>{{end}}
```
Comment (wxiaoguang) [inline templates/mail/issue/default.tmpl]: There is no `ctx` support in mail renderer.

**Review by wxiaoguang — DISMISSED** [review]
Still need more fixes

**Comment by delvh** [comment]
So effectively, I should revert the changes in the mail directory?

**Review by puni9869 — APPROVED** [review]

**Pull request merged**